st2api/st2api/controllers/v1/actionexecutions.py

```diff
@@ -153,7 +155,16 @@ def _schedule_execution(self, liveaction):
             liveaction_db.parameters = param_utils.render_live_params(
                 runnertype_db.runner_parameters, action_db.parameters, liveaction_db.parameters,
                 liveaction_db.context)
-        except ParamException as e:
+        except ParamException:
+            LOG.info('ParamException')
+            # By this point the execution is already in the DB therefore need to mark it failed.
+            _, e, tb = sys.exc_info()
+            action_service.update_status(
+                liveaction=liveaction_db,
+                new_status=LIVEACTION_STATUS_FAILED,
+                result={'error': str(e), 'traceback': ''.join(traceback.format_tb(tb, 20))})
```

Comment:
Potential future code improvement - it would be nice if we had a single place / function where the result object for error scenarios is constructed.

If it's constructed in multiple places it's hard to keep it in sync.